### PR TITLE
Re-index existing articles

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This uses the [Data Update Scripts](https://docs.dev.to/backend/data-update-scripts/) to re-index the existing articles with the approved attribute.

This PR ties into [#6032](https://github.com/thepracticaldev/dev.to/pull/6032) where I've added the approved attribute.

## Related Tickets & Documents
This PR ties into [#6032](https://github.com/thepracticaldev/dev.to/pull/6032) where I've added the approved attribute.

[Issue #372](https://github.com/thepracticaldev/tech-private/issues/372)

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by rhymes** [comment]
@mstruve looking at this PR I noticed a potential to break future builds. For example, in this case we're committing a script that triggers a method that's likely to disappear soon (but the reasoning is valid for all methods). 

If `.trigger_index` gets removed what would happen? Would we remove the script? If we'd do so, we'd break the correspondence between the DB table and the folder.

The only solution I'm thinking is to make script no ops if we end up removing methods that are called in those scripts.

Let me know if any of this doesn't make sense, as it's a hunch I just had

File: spec/lib/data_update_scripts/re_index_existing_articles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require "rails_helper"
+require Rails.root.join("lib/data_update_scripts/20200217131245_re_index_existing_articles_with_approved.rb")
+
+describe DataUpdateScripts::ReIndexExistingArticlesWithApproved do
+  it "triggers a re-index on articles" do
+    expect(Article).to respond_to(:trigger_index)
```
Comment (rhymes) [inline spec/lib/data_update_scripts/re_index_existing_articles_spec.rb]: I think this is only testing if the article object has the method `trigger_index`, but it's not testing if running script triggers that method.

As you're loading article objects from the DB I don't even think there's an easy way to test this. You might want to test it like this:

```ruby
create(:article)
described_class.new.run
# check there are the proper enqueued jobs
```

Comment (Ridhwana) [reply spec/lib/data_update_scripts/re_index_existing_articles_spec.rb]: @rhymes I kept this purposefully simple after chatting with @mstruve, the most important part of the script was to test that when we remove Algolia from the Article model we would know to remove the data update. This code isn't going to be around for long and I didn't want to spend too much time on it

Comment (rhymes) [reply spec/lib/data_update_scripts/re_index_existing_articles_spec.rb]: makes sense!

Comment (Ridhwana) [reply spec/lib/data_update_scripts/re_index_existing_articles_spec.rb]: Thanks @rhymes!

File: lib/data_update_scripts/20200217131245_re_index_existing_articles_with_approved.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module DataUpdateScripts
+  class ReIndexExistingArticlesWithApproved
+    def run
+      Article.find_each { |article| Article.trigger_index(article, false) }
```
Comment (benhalpern) [inline lib/data_update_scripts/20200217131245_re_index_existing_articles_with_approved.rb]:
```suggestion
      Article.published.find_each { |article| Article.trigger_index(article, false) }
```

We don't need to loop through unpublished stuff eh?

Comment (Ridhwana) [reply lib/data_update_scripts/20200217131245_re_index_existing_articles_with_approved.rb]: hmmm, yeah that would make sense.

File: spec/lib/data_update_scripts/re_index_existing_articles_spec.rb (hunk 1 above)
Comment (mstruve) [reply spec/lib/data_update_scripts/re_index_existing_articles_spec.rb]: Would be kinda nice to just have a simple framework that runs all of the data update scripts before the test suite, kinda like migrations, and just ensures we dont error out. https://github.com/thepracticaldev/dev.to/projects/9#card-33234638

**Pull request merged**